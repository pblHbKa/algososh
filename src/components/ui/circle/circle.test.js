import renderer from "react-test-renderer";
import { ElementStates } from "../../../types/element-states";

import { Circle } from "./circle";

it("Кружок без буквы отрисовывается правильно", () => {
  const circle = renderer.create(<Circle />).toJSON();
  expect(circle).toMatchSnapshot();
});

it("Кружок с буквами отрисовывается правильно", () => {
  const circle = renderer.create(<Circle letter="test" />).toJSON();
  expect(circle).toMatchSnapshot();
});

it("Кружок с head отрисовывается правильно", () => {
  const circle = renderer.create(<Circle head={"test"} />).toJSON();
  expect(circle).toMatchSnapshot();
});

it("Кружок с react-элементом в head отрисовывается правильно", () => {
  const circle = renderer.create(<Circle head={<div>test</div>} />).toJSON();
  expect(circle).toMatchSnapshot();
});

it("Кружок с tail отрисовывается правильно", () => {
  const circle = renderer.create(<Circle tail={"test"} />).toJSON();
  expect(circle).toMatchSnapshot();
});

it("Кружок с react-элементом в tail отрисовывается правильно", () => {
  const circle = renderer.create(<Circle tail={<div>test</div>} />).toJSON();
  expect(circle).toMatchSnapshot();
});

it("Маленький кружок отрисовывается правильно", () => {
  const circle = renderer.create(<Circle isSmall={true} />).toJSON();
  expect(circle).toMatchSnapshot();
});

it("Кружок в обычном состоянии отрисовывается правильно", () => {
  const circle = renderer
    .create(<Circle state={ElementStates.Default} />)
    .toJSON();
  expect(circle).toMatchSnapshot();
});

it("Кружок в состоянии изменения отрисовывается правильно", () => {
  const circle = renderer
    .create(<Circle state={ElementStates.Changing} />)
    .toJSON();
  expect(circle).toMatchSnapshot();
});

it("Обработанный кружок отрисовывается правильно", () => {
  const circle = renderer
    .create(<Circle state={ElementStates.Modified} />)
    .toJSON();
  expect(circle).toMatchSnapshot();
});
