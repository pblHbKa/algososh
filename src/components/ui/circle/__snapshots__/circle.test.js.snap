// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Кружок без буквы отрисовывается правильно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок в обычном состоянии отрисовывается правильно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок в состоянии изменения отрисовывается правильно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок с head отрисовывается правильно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    test
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок с react-элементом в head отрисовывается правильно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div>
      test
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Кружок с react-элементом в tail отрисовывается правильно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div>
      test
    </div>
  </div>
</div>
`;

exports[`Кружок с tail отрисовывается правильно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    test
  </div>
</div>
`;

exports[`Кружок с буквами отрисовывается правильно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      test
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Маленький кружок отрисовывается правильно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Обработанный кружок отрисовывается правильно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
